### PR TITLE
Feat(eos_cli_config_gen): Add `route_map_in/out` for `router_bgp.address_family_evpn.neighbors[]`

### LINKED ISSUE #4458: Feat (eos_cli_config_gen): Add support for route-map in|out under router bgp address-family evpn
### Enhancement summary

add support for route-map in or out per bgp neighbor under router_bgp.address_family_evpn:

```
router_bgp:

  address_family_evpn:
    neighbors:
      - ip_address: <str; required; unique>
        activate: <bool>

        # Inbound route-map name.
        route_map_in: <str>

        # Outbound route-map name.
        route_map_out: <str>
```

### Which component of AVD is impacted

eos_cli_config_gen

### Use case example

```
router_bgp:

  address_family_evpn:
    neighbors:
      - ip_address: 10.100.252.1
        activate: true

        # Inbound route-map name.
        route_map_in: RM_Example_IN

        # Outbound route-map name.
        route_map_out: RM_Example_OUT
```

```
router bgp 65001
 address-family evpn
   neighbor 10.100.252.1 route-map RM_Example_IN in
   neighbor 10.100.252.1 route-map RM_Example_OUT out
```

### Describe the solution you would like

```
router_bgp:

  address_family_evpn:
    neighbors:
      - ip_address: <str; required; unique>
        activate: <bool>

        # Inbound route-map name.
        route_map_in: <str>

        # Outbound route-map name.
        route_map_out: <str>
```

### Describe alternatives you have considered

eos_cli or cvp configlet

### Additional context

_No response_

### Contributing Guide

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by laxmikantchintakindi**
## Change Summary

Add `route_map_in/out` for `router_bgp.address_family_evpn.neighbors[]`

## Related Issue(s)

Fixes #4458

## Component(s) name

`arista.avd.eos_cli_config_gen`

## Proposed changes
Add support for route-map in or out per bgp neighbor under router_bgp.address_family_evpn:
```
router_bgp:

  address_family_evpn:
    neighbors:
      - ip_address: <str; required; unique>
        activate: <bool>

        # Inbound route-map name.
        route_map_in: <str>

        # Outbound route-map name.
        route_map_out: <str>
```
### User Checklist

- N/A

### Repository Checklist

- [x] My code has been rebased from devel before I start
- [x] I have read the [**CONTRIBUTING**](https://avd.sh/en/latest/docs/contribution/overview.html) document.
- [ ] My change requires a change to the documentation and documentation have been updated accordingly.
- [x] I have updated [molecule CI](https://github.com/aristanetworks/avd/tree/devel/ansible_collections/arista/avd/molecule) testing accordingly. (check the box if not applicable)

### CODE REVIEW

File: python-avd/pyavd/_eos_cli_config_gen/j2templates/eos/router-bgp.j2 [hunk 1]
```diff
@@ -896,6 +896,12 @@ router bgp {{ router_bgp.as }}
 {%             if neighbor.additional_paths.receive is arista.avd.defined(true) %}
       neighbor {{ neighbor.ip_address }} additional-paths receive
 {%             endif %}
+{%             if neighbor.route_map_in is arista.avd.defined %}
+      neighbor {{ neighbor.ip_address }} route-map in
```
Comment (Shivani-gslab) [inline python-avd/pyavd/_eos_cli_config_gen/j2templates/eos/router-bgp.j2]:
```suggestion
      neighbor {{ neighbor.ip_address }} route-map {{ neighbor.route_map_in }} in
```

File: python-avd/pyavd/_eos_cli_config_gen/j2templates/eos/router-bgp.j2 [hunk 2]
```diff
@@ -896,6 +896,12 @@ router bgp {{ router_bgp.as }}
...
+{%             endif %}
+{%             if neighbor.route_map_out is arista.avd.defined %}
+      neighbor {{ neighbor.ip_address }} route-map out
```
Comment (Shivani-gslab) [inline python-avd/pyavd/_eos_cli_config_gen/j2templates/eos/router-bgp.j2]:
```suggestion
      neighbor {{ neighbor.ip_address }} route-map {{ neighbor.route_map_out }} out
```

File: ansible_collections/arista/avd/molecule/eos_cli_config_gen/inventory/host_vars/router-bgp-evpn.yml [hunk 3]
```diff
@@ -60,6 +60,8 @@ router_bgp:
       missing_policy:
         direction_out:
           action: deny-in-out
+      route_map_in: RM1
+      route_map_out: RM2
```
Comment (Shivani-gslab) [inline ansible_collections/arista/avd/molecule/eos_cli_config_gen/inventory/host_vars/router-bgp-evpn.yml]: You are adding the test case at wrong place, it needs to be added under `address_family_evpn.neighbors`

**Review by Shivani-gslab — CHANGES_REQUESTED** [review]

File: ansible_collections/arista/avd/molecule/eos_cli_config_gen/inventory/host_vars/router-bgp-evpn.yml (hunk 3 above)
Comment (laxmikantchintakindi) [reply ansible_collections/arista/avd/molecule/eos_cli_config_gen/inventory/host_vars/router-bgp-evpn.yml]: Fixed.

**Review by Shivani-gslab — APPROVED** [review]

**Review by MaheshGSLAB — APPROVED** [review]

**Review by ClausHolbechArista — DISMISSED** [review]
Blocking until 5.0 is out

File: python-avd/pyavd/_eos_cli_config_gen/j2templates/eos/router-bgp.j2 [hunk 4]
```diff
@@ -896,6 +896,12 @@ router bgp {{ router_bgp.as }}
 {%             if neighbor.additional_paths.receive is arista.avd.defined(true) %}
       neighbor {{ neighbor.ip_address }} additional-paths receive
 {%             endif %}
+{%             if neighbor.route_map_in is arista.avd.defined %}
+      neighbor {{ neighbor.ip_address }} route-map {{ neighbor.route_map_in }} in
+{%             endif %}
+{%             if neighbor.route_map_out is arista.avd.defined %}
+      neighbor {{ neighbor.ip_address }} route-map {{ neighbor.route_map_out }} out
+{%             endif %}
```
Comment (gmuloc) [inline python-avd/pyavd/_eos_cli_config_gen/j2templates/eos/router-bgp.j2:904]: please add these fields in documentation template (for neighbors and for peer groups) to be consistent with other address families (BGP LU, SRTE,...)

You will need to molecule a lot of things

**Review by gmuloc — CHANGES_REQUESTED** [review]

File: python-avd/pyavd/_eos_cli_config_gen/j2templates/eos/router-bgp.j2 (hunk 4 above)
Comment (laxmikantchintakindi) [reply python-avd/pyavd/_eos_cli_config_gen/j2templates/eos/router-bgp.j2:904]: Added for evpn neighbors and peer groups.

**Review by gmuloc — APPROVED** [review]

**Comment by kmueller68** [comment]
Thank you very much for implementing it. 👍

**Pull request merged**